lib/workers/branch/schedule.spec.ts

```diff
@@ -257,5 +257,19 @@ describe(getName(), () => {
       const res = schedule.isScheduledNow(config);
       expect(res).toBe(false);
     });
+    it('supports weekday instances', () => {
+      config.schedule = ['on Monday on the first day instance'];
+
+      const cases: [string, boolean][] = [
+        ['2017-02-01T06:00:00.000', false], // Locally Thursday, 2 February 2017 6am
+        ['2017-02-06T06:00:00.000', true], // Locally Monday, 6 February 2017 6am
+        ['2017-02-13T06:00:00.000', false], // Locally Monday, 13 February 2017 6am
+      ];
+
+      cases.forEach(([datetime, expected]) => {
+        mockDate.set(datetime);
+        expect(schedule.isScheduledNow(config)).toBe(expected);
+      });
+    });
```

Comment:
Can you add a test for `before 3am on Monday` in timezone `Asia/Tokyo` (`UTC+0900`)

When run ~2am it should match. So this can potentially also fix the discussed issue?


ref https://github.com/renovatebot/renovate/discussions/9724

Reply:
Added a testcase, and yes this fixes that issue as well.